protocol/builder-separation.md

```diff
@@ -0,0 +1,115 @@
+
+**Deny DOS Vectors**.
+The Sequencer should not be newly exposed to mechanisms which could be abused to reduce performance.
+
+**Support Fallback**.
+When Block Builders fail to deliver valid blocks, it should not impact the service liveness, and the "traditionally" built block should be usable in its place.
+
+**Be Asynchronous**.
+The Sequencer should not take Block Builders as a blocking source of latency.
+
+# Proposed Solution
+We should modify our Sequencing to support an event-driven, asyncrhonous mechanism for discovering and publishing blocks. These modifications should be codified as spec, and the API interfaces between the Sequencer and Builders should be especially defined.
+
+Recently, the Derivation package has been refactored to use an event-driven architecture. This work was defined in the previous "Derivers" [design document](https://github.com/ethereum-optimism/design-docs/pull/1).
+
+In this event-based architecture, the Sequencer regularly emits events to either:
+* Begin block building
+* Seal the block (end block building)
+* FCU the block (to make it cannonical against the Execution Engine)
+
+To support PBS, the Sequencer is newly responsible for managing the flow of a Block Builder as an outside agent. To accomplish this, it should emit timing-based calls which tell the block builder critical information needed to provide blocks, including block attributes and timing information.
+
+## High Level Overview
+Here is a diagram of the components and their interactions.
+```mermaid
+sequenceDiagram
+    participant Customer
+    participant ProxyD
+    participant Block Builder
+    participant Execution Engine
+    participant Sequencer
+    participant Replicas
+
+    Customer->>+ProxyD: Send Transaction
+    ProxyD->>+Block Builder: Forward Transaction
+    ProxyD->>+Execution Engine: Forward Transaction
+
+        Note over Sequencer: On a regular basis...
+    Sequencer->>Block Builder: Start Building <Attributes>
+    Sequencer->>Execution Engine: Start Building <Attributes>
+        Note over Sequencer: Time Passes
+
+    Sequencer->>Block Builder: Notify of Building Deadline
+    Block Builder->>Sequencer: Deliver Block Asynchronously
+        Note over Sequencer: Block Building Deadline Passes
+    Sequencer->>Execution Engine: Seal Block
+        Note over Sequencer: Pick which block should be used
+    Sequencer->>Execution Engine: FCU Block onto Execution Engine
+    Sequencer->>Replicas: Publish Block
+```
+To break down the components of this diagram:
+
+**Transactions are sent to the Execution Engine and Block Builder via Proxyd**.  Today, the Execution Engine is responsible for being informed of Transactions and maintaining a mempool. With PBS, the Block Builder would have the same responsibilities. The Sequencer *does not* forward Transactions. This is because the Sequencer is not involved in the block building activities directly, and does not benefit from acting as a pass-through
```

Comment:
We don't necessarily need to be opinionated about the orderflow. This architecture makes sense for an OP Mainnet style network, but you could imagine many architectures around the order flow

Reply:
I agree, referencing `proxyd` here is an over-constraint of the design which makes it OP Mainnet specific. The purpose of calling that mechanism out is:
* `proxyd` *is* the reference standard for delivering transactions to Execution Engines today, and the way in which transactions are delivered to Execution Engines should be matched when delivering to Block Builders.
* Sequencers `op-node` should not have to process or route transaction traffic to other components.

I will expand on this to make it clear, though it may still be useful to reference `proxyd` as the entry-point for customer transactions.